src/hooks/street-view-panorama.ts

```diff
@@ -0,0 +1,72 @@
+import {useMemo, useState} from 'react';
+
+import {useGoogleMap} from './map-instance';
+
+interface StreetViewProps {
+  divElement?: HTMLElement | null;
+  position?: google.maps.LatLng | google.maps.LatLngLiteral;
+  heading?: number;
+  pitch?: number;
+  zoom?: number;
+}
+/**
+ * Hook to get Street View Panorama
+ */
+export const useStreetView = (
+  props: StreetViewProps
+): google.maps.StreetViewPanorama | null => {
+  const {divElement, position, heading, pitch, zoom} = props;
+
+  // Get map instance
+  const {map} = useGoogleMap();
+
+  const [panorama, setPanorama] =
+    useState<google.maps.StreetViewPanorama | null>(null);
+
+  // Creates a Street View instance
+  const streetViewPanorama =
+    useMemo<google.maps.StreetViewPanorama | null>(() => {
+      // Wait for Google Maps API to be initialized
+      if (!map) {
+        return null;
+      }
+
+      if (!divElement) {
+        const newPanorama = map.getStreetView();
+        newPanorama.setPov({heading, pitch});
```

Comment:
`heading` and `pitch` are optional, right? So this should only be done when data was passed.

Reply:
Made them optional now. Just another question that came up with this:

Is it possible to set one without the other? Like only setting `heading` without `pitch`? If not, we could also think of passing `pov` as prop.

Reply:
I think it's possible to only use one of them, because according to the documentation both have a default value.